cell_classification/segmentation_data_prep_test.py

```diff
@@ -33,9 +86,143 @@ def test_get_image():
         assert not np.array_equal(CD8_img, CD4_img)
 
 
+def prepare_test_data_folders(
+    num_folders, temp_dir, selected_markers, random=False, scale=1.0
+):
+    data_folders = []
+    for i in range(num_folders):
+        folder = os.path.join(temp_dir, "fov_1" + str(i))
+        os.mkdir(folder)
+        data_folders.append(folder)
+        for marker in selected_markers:
+            if random:
+                img = np.random.rand(256, 256) * scale
+            else:
+                img = np.ones([256, 256])
+            imwrite(
+                os.path.join(temp_dir, "fov_1" + str(i), marker + ".tiff"),
+                img,
+            )
+    return data_folders
+
+
+def test_calculate_normalization_matrix():
+    # instantiate data_prep and conversion_matrix
+    data_prep = prep_object()
+    conversion_matrix = prepare_conversion_matrix()
+    # get markers from there
+    selected_markers = list(conversion_matrix.columns)
+    selected_markers.remove("cluster_labels")
+    with tempfile.TemporaryDirectory() as temp_dir:
+        # create temporary folders with data
+        data_folders = prepare_test_data_folders(5, temp_dir, selected_markers)
+        norm_dict = data_prep.calculate_normalization_matrix(
+            data_folders=data_folders,
+            normalization_dict_path=os.path.join(temp_dir, "norm_dict_test.json"),
+            normalization_quantile=0.99,
+            selected_markers=selected_markers,
+        )
+        norm_dict_loaded = json.load(
+            open(os.path.join(temp_dir, "norm_dict_test.json"))
+        )
+        # check if the normalization_dict is correctly written to the json file
+        assert norm_dict_loaded == norm_dict
+        # check if the normalization_dict has the correct values
+        for marker in norm_dict.keys():
+            assert norm_dict[marker] == 1.0
+        # check if the normalization_dict has the correct keys
+        for marker in selected_markers:
+            assert marker in norm_dict.keys()
+        # check if the normalization_dict works correctly when
+        # normalization_dict_path is given to init
+        conversion_matrix = prepare_conversion_matrix()
+        conversion_matrix_path = os.path.join(temp_dir, "conversion_matrix.csv")
```

Comment:
There's a lot of code here, and above, for making the `conversion_matrix`. However, this conversion matrix isn't needed for the function being tested here, only to initialize the class. This is a hint that the logic of things are being set up could be improved. 

Either 1) Create a dummy version of the SegmentationTFRecords object that can be initialized without requiring all of the arguments to be specified, to use just for testing, or 
2) Move the logic that is currently being executed in the `init` portion of the class into a dedicated helper function that is called later, so it doesn't need to be run every time

Reply:
Put stuff form `init` into `make_tf_record` because it's needed there but not in `init`